### ISSUE TITLE
[1.11.2-1.0] Concrete solidifying differently from vanilla

### ISSUE THREAD

**Issue by starg09** [D-prog-55/ConcreteMod#1]
This is a bit minor, but there's a difference between how your mod solidifies concrete respect to the mechanic in 1.12 Vanilla. Maybe you could adjust it? :)

* [How it works on your mod](https://gfycat.com/RecklessHotDogfish)
* [How it works on vanilla](https://gfycat.com/PlumpExcitableAss)

As you can see, on vanilla the concrete never solidifies if water is touching the block's bottom. Only works from the sides and above (which I didn't record, but can confirm works like that as well).

**Comment by D-prog-55** [comment]
Well, when I see the "Vanilla" gif, placing a concrete powder block with water just below looks more of a bug (because hey, it's a snapshot) rather than something Mojang actually wants (it _could_ be, but I really doubt about it), because it insta-tps one block down. Plus, I wrote a **HUGE** "if(water is up||water is down||...)" like statement, so it's not something that would be really hard to change.

BTW it's not realistic if you have a thin layer of water under the block, so I could maybe check for the AABB of the falling block being in water instead of checking if below is a water block (that's the only case where it would matter as the other cases always touch water).

And also, the entity will only turn solid if in water (and not if water is below, which is a bit contradictory).

So yeah, maybe I could do it, but I prefer waiting for the official release to make modifications if I need to (even if 17w13a confirms what you said).

**Comment by starg09** [comment]
That sounds reasonable, if you consider it to be a bug in vanilla.

Just as a thought I just had: based on what you're saying on how it wouldn't be really hard to change it, maybe you could implement a "use vanilla behaviour" (off by default, with a note about it posibly being buggy), which, when enabled, would remove that specific case?

**Comment by D-prog-55** [comment]
> maybe you could implement a "use vanilla behaviour"

That sounds like a good idea, now on my todo list ;p.

**Comment by D-prog-55** [comment]
Double post (but it's my Github project so I don't care ;p):

Done with the vanilla behavior. It looks like it "setblock"s a concrete powder one block down and then it gets solidified by nearby water... if there still is (tested in 17w16b).

[GIF](http://dbrown55.ddns.net:8000/files/concrete.gif)

(Notes: 1/This is Windows 7 and 2/Self-hosted server ^^)

**Comment by starg09** [comment]
Oh, nice! Good to know, wasn't aware of that condition of having water around 😄 

Thanks for taking a look into this. As a note, since chisel also backported this feature recently we're currently testing without your mod, due to the overlap.

So may not be able to report much in the future :/

**Comment by D-prog-55** [comment]
Yeah I know, I made a chisel<->my mod recipe for the concrete. Kinda sucks that Chisel now has it too but...

[Another pic on my own webserv ^^](dbrown55.ddns.net:8000/files/whyPeopleCouldStillUseMyMod.png)
